src/test/java/com/descope/sdk/mgmt/impl/OutboundAppsByTokenServiceImplTest.java

```diff
@@ -0,0 +1,150 @@
+package com.descope.sdk.mgmt.impl;
+
+import static com.descope.literals.Routes.ManagementEndPoints.MANAGEMENT_FETCH_OUTBOUND_APP_TENANT_TOKEN;
+import static com.descope.literals.Routes.ManagementEndPoints.MANAGEMENT_FETCH_OUTBOUND_APP_TENANT_TOKEN_LATEST;
+import static com.descope.literals.Routes.ManagementEndPoints.MANAGEMENT_FETCH_OUTBOUND_APP_USER_TOKEN;
+import static com.descope.literals.Routes.ManagementEndPoints.MANAGEMENT_FETCH_OUTBOUND_APP_USER_TOKEN_LATEST;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.descope.exception.DescopeException;
+import com.descope.exception.ServerCommonException;
+import com.descope.model.client.Client;
+import com.descope.model.outbound.FetchLatestOutboundAppUserTokenRequest;
+import com.descope.model.outbound.FetchOutboundAppTenantTokenRequest;
+import com.descope.model.outbound.FetchOutboundAppTenantTokenResponse;
+import com.descope.model.outbound.FetchOutboundAppUserTokenRequest;
+import com.descope.model.outbound.FetchOutboundAppUserTokenResponse;
+import com.descope.proxy.ApiProxy;
+import com.descope.proxy.impl.ApiProxyBuilder;
+import java.util.Arrays;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+
+class OutboundAppsByTokenServiceImplTest {
```

Comment:
Java is a bit weird in that we actually do end-to-end integration tests right here in the unit tests. Can you add these tests right here?